packages/frontend/@n8n/design-system/src/components/N8nInlineTextEdit/InlineTextEdit.vue

```diff
@@ -0,0 +1,197 @@
+<script lang="ts" setup>
+import { useElementSize } from '@vueuse/core';
+import { EditableArea, EditableInput, EditablePreview, EditableRoot } from 'reka-ui';
+import { computed, ref, useTemplateRef, watch } from 'vue';
+
+type Props = {
+	modelValue: string;
+	readOnly?: boolean;
+	maxLength?: number;
+	maxWidth?: number;
+};
+
+const props = withDefaults(defineProps<Props>(), {
+	readOnly: false,
+	maxLength: 100,
+	maxWidth: 200,
+});
+
+const emit = defineEmits<{
+	'update:model-value': [value: string];
+}>();
+
+const editableRoot = useTemplateRef('editableRoot');
+
+function forceFocus() {
+	if (editableRoot.value && !props.readOnly) {
+		editableRoot.value.edit();
+	}
+}
+
+function forceCancel() {
+	if (editableRoot.value) {
+		newValue.value = props.modelValue;
+		editableRoot.value.cancel();
+	}
+}
+
+defineExpose({ forceFocus, forceCancel });
+
+function onSubmit() {
+	if (newValue.value === '') {
+		newValue.value = props.modelValue;
+		temp.value = props.modelValue;
+		return;
+	}
+	emit('update:model-value', newValue.value);
+}
+
+function onInput(value: string) {
+	newValue.value = value;
+	measureText.value = value || 'A';
+}
+
+function onStateChange(state: string) {
+	if (state === 'cancel') {
+		temp.value = newValue.value;
+	}
+}
+
+// Resize logic
+const newValue = ref(props.modelValue);
+const temp = ref(props.modelValue);
+const measureText = ref(props.modelValue || 'A');
+
+const measureSpan = useTemplateRef('measureSpan');
+const { width: measuredWidth } = useElementSize(measureSpan);
+
+const inputWidth = computed(() => {
+	const minWidth = 60; // Minimum width
```

Comment:
The `minWidth` should also be configurable via props and we should make it a bit bigger (64?). 

The default text is cut off: 
<img width="290" alt="image" src="https://github.com/user-attachments/assets/b12d0eba-7671-4f0f-ad5b-5a2136ca301e" />